packages/ai/core/tool/tool.ts

```diff
@@ -94,28 +96,28 @@ The arguments for configuring the tool. Must match the expected arguments define
 Helper function for inferring the execute args of a tool.
  */
 // Note: special type inference is needed for the execute function args to make sure they are inferred correctly.
-export function tool<PARAMETERS extends Parameters, RESULT>(
+export function tool<PARAMETERS extends Parameters, RESULT extends Parameters>(
```

Comment:
`RESULT extends Parameters` seems strange.

Reply:
`generateCode` requires an returns `zodSchema` so LLM can understand whats the output of each tools.

It'll help LLM to create variables to save the tool-calling-result and pass it to next tool. 

Reason why i used `Parameters` type is to strictly specify execute tool according to its `returns: zodSchema`

I could rework this as

```ts
type Returns = Parameters // same properties

export function tool<PARAMETERS extends Parameters, RESULT, RETURNS extends Returns>(
tool: CoreTool<PARAMETERS, RETURNS> & {
    execute: (
      args: inferParameters<PARAMETERS>,
      options: ToolExecutionOptions,
    ) => PromiseLike<inferParameters<RETURNS>>;
  }
)
```